### ISSUE TITLE
Link for gnupru is 2021.10 release is dead

### ISSUE THREAD

**Issue by VedantParanjape** [dinuxbg/gnupru#42]
http://dinux.eu/gnupru/binutils-2.37.20211017.tar.bz2 This link is dead in the download-and-prepare.sh script in the 2021.10 release.

**Comment by dinuxbg** [comment]
The link works for me. Can you please use command line and share the log from the download failure?

Please consider switching to the [officially released Binutils 2.38 package](http://ftpmirror.gnu.org/binutils/binutils-2.38.tar.bz2). The `gnupru` master branch already did the switch in commit 105dcf5.

**Comment by VedantParanjape** [comment]
```
gzip: /gnupru-2021.10/src/binutils-gdb.tar.gz: No such file or directory
--2022-04-10 11:09:54--  http://dinux.eu/gnupru/binutils-2.37.20211017.tar.bz2
Resolving dinux.eu (dinux.eu)... 193.107.36.199
Connecting to dinux.eu (dinux.eu)|193.107.36.199|:80... ^C
```
also trying to wget it, just keeps on connecting.

```
wget http://dinux.eu/gnupru/binutils-2.37.20211017.tar.bz2                                                                                                                
--2022-04-10 23:00:03--  http://dinux.eu/gnupru/binutils-2.37.20211017.tar.bz2
Resolving dinux.eu (dinux.eu)... 193.107.36.199
Connecting to dinux.eu (dinux.eu)|193.107.36.199|:80...
```

**Comment by VedantParanjape** [comment]
> The link works for me. Can you please use command line and share the log from the download failure?
> 
> Please consider switching to the [officially released Binutils 2.38 package](http://ftpmirror.gnu.org/binutils/binutils-2.38.tar.bz2). The `gnupru` master branch already did the switch in commit [105dcf5](https://github.com/dinuxbg/gnupru/commit/105dcf5f0e360c6ea832cc5ec99dbe33a4652237).

I can't use the latest release, or that commit as it bumps to gcc-12, which has the new version of pru firmware. I haven't yet ported my apps to use the new firmware. Is there a work around ?

**Comment by dinuxbg** [comment]
As a workaround, change the link in `download-and-prepare.sh` to :

```
BINUTILS_URL=http://ftpmirror.gnu.org/binutils/binutils-2.38.tar.bz2
```

My snapshot and 2.38 are functionally the same for the PRU target.

**Comment by dinuxbg** [comment]
What error do you get from `wget` ? Do you open http://dinux.eu with your browser?

I have not blocked any traffic, for any reason.

**Comment by VedantParanjape** [comment]
> What error do you get from `wget` ? Do you open http://dinux.eu with your browser?
> 
> I have not blocked any traffic, for any reason.

As for wget, it keeps retrying until I kill it. I cannot open the site in browser. It keeps on loading forever. Maybe the ISP flagged your domain ?

**Comment by VedantParanjape** [comment]
> As a workaround, change the link in `download-and-prepare.sh` to :
> 
> ```
> BINUTILS_URL=http://ftpmirror.gnu.org/binutils/binutils-2.38.tar.bz2
> ```
> 
> My snapshot and 2.38 are functionally the same for the PRU target.

I am using the script in an GitHub action, can you update the old release if possible ?

**Comment by dinuxbg** [comment]
> > What error do you get from `wget` ? Do you open http://dinux.eu with your browser?
> > I have not blocked any traffic, for any reason.
> 
> As for wget, it keeps retrying until I kill it. I cannot open the site in browser. It keeps on loading forever. Maybe the ISP flagged your domain ?

I did not see any errors in my server logs. https://isitup.org/dinux.eu also lists the site as up and running.

**Comment by VedantParanjape** [comment]
> > > What error do you get from `wget` ? Do you open http://dinux.eu with your browser?
> > > I have not blocked any traffic, for any reason.
> > 
> > 
> > As for wget, it keeps retrying until I kill it. I cannot open the site in browser. It keeps on loading forever. Maybe the ISP flagged your domain ?
> 
> I did not see any errors in my server logs. https://isitup.org/dinux.eu also lists the site as up and running.

Okay, I understood the issue. The link is `http`, whereas the isitup.org has a `https`

**Comment by dinuxbg** [comment]
> > The link works for me. Can you please use command line and share the log from the download failure?
> > Please consider switching to the [officially released Binutils 2.38 package](http://ftpmirror.gnu.org/binutils/binutils-2.38.tar.bz2). The `gnupru` master branch already did the switch in commit [105dcf5](https://github.com/dinuxbg/gnupru/commit/105dcf5f0e360c6ea832cc5ec99dbe33a4652237).
> 
> I can't use the latest release, or that commit as it bumps to gcc-12, which has the new version of pru firmware. I haven't yet ported my apps to use the new firmware. Is there a work around ?

You should be able to use the latest GCC/Binutils with any PRU firmware. I would suggest to "bump to gcc-12" :)

In the [latest release page](https://github.com/dinuxbg/gnupru/releases), you can also download prebuilt toolchain binaries for Linux x86_64 host, Linux armhf host, and Windows hosts.

FYI, after GCC-12 is officially released, I intend to make a corresponding `gnupru` release.

**Comment by VedantParanjape** [comment]
> > > The link works for me. Can you please use command line and share the log from the download failure?
> > > Please consider switching to the [officially released Binutils 2.38 package](http://ftpmirror.gnu.org/binutils/binutils-2.38.tar.bz2). The `gnupru` master branch already did the switch in commit [105dcf5](https://github.com/dinuxbg/gnupru/commit/105dcf5f0e360c6ea832cc5ec99dbe33a4652237).
> > 
> > 
> > I can't use the latest release, or that commit as it bumps to gcc-12, which has the new version of pru firmware. I haven't yet ported my apps to use the new firmware. Is there a work around ?
> 
> You should be able to use the latest GCC/Binutils with any PRU firmware. I would suggest to "bump to gcc-12" :)
> 
> In the [latest release page](https://github.com/dinuxbg/gnupru/releases), you can also download prebuilt toolchain binaries for Linux x86_64 host, Linux armhf host, and Windows hosts.

I took a look at it, could you please create a debian release instead ? or an install script.
> 
> FYI, after GCC-12 is officially released, I intend to make a corresponding `gnupru` release.

**Comment by dinuxbg** [comment]
> > > > The link works for me. Can you please use command line and share the log from the download failure?
> > > > Please consider switching to the [officially released Binutils 2.38 package](http://ftpmirror.gnu.org/binutils/binutils-2.38.tar.bz2). The `gnupru` master branch already did the switch in commit [105dcf5](https://github.com/dinuxbg/gnupru/commit/105dcf5f0e360c6ea832cc5ec99dbe33a4652237).
> > > 
> > > 
> > > I can't use the latest release, or that commit as it bumps to gcc-12, which has the new version of pru firmware. I haven't yet ported my apps to use the new firmware. Is there a work around ?
> > 
> > 
> > You should be able to use the latest GCC/Binutils with any PRU firmware. I would suggest to "bump to gcc-12" :)
> 
> From the last discussion, I thought there was a breakage. I'll bump to gcc-12 if that's the case. Still

The breakage is in the remoteproc/RPMSG firmware.
 - PRU firmware for kernel 5.10 requires new (GCC 12) toolchain.
 - PRU firmware for kernel 4.19 can be built with both new and old toolchains.
 
> 
> > In the [latest release page](https://github.com/dinuxbg/gnupru/releases), you can also download prebuilt toolchain binaries for Linux x86_64 host, Linux armhf host, and Windows hosts.
> 
> I took a look at it, could you please create a debian release instead ? or an install script.

RCN's  repo has debian packages for armhf hosts.

The gnupru binary releases are simple to install, so I don't see a point in an install script:
  - Download.
  - Extract.
  - Add the `bin` sub-directory to your `PATH`

**Comment by VedantParanjape** [comment]
Thanks @dinuxbg Issue solved for me
